### PR TITLE
FIX deprecation warnings in test_cli

### PR DESCRIPTION
**Pull request by tomMoral**
Deprecations reported in #182 :

- [x] `autocompletion` -> `shell_complete`
- [ ] `Creating a LegacyVersion has been deprecated and will be removed in the next major release`
- [x] User warning on editable config.

### CODE REVIEW

**Comment by tomMoral** [comment]
The last one with the `LegacyVersion` seems to be due to a bug in `setuptools` that have been fixed in : pypa/setuptools#2889

File: benchopt/cli/main.py [hunk 1]
```diff
@@ -36,20 +36,20 @@
...
               " are included. To include multiple solvers, "
-              "use multiple `-s` options.", autocompletion=get_solvers)
+              "use multiple `-s` options.", shell_complete=get_solvers)
 @click.option('--force-solver', '-f', 'forced_solvers',
               metavar="<solver_name>", multiple=True, type=str,
               help="Force the re-run for <solver_name>. This "
               "avoids caching effect when adding a solver. "
               "To select multiple solvers, use multiple `-f` options.",
-              autocompletion=get_solvers)
+              shell_complete=get_solvers)
```
Comment (mathurinm) [inline benchopt/cli/main.py:45]: Through this PR I realize that there may be some autocompletion working? But hitting tab after having typed a solver name does not do anything for me, e.g. in this situation:

`(base) ➜  lasso git:(main) ✗ benchopt run . -s sk`

**Review by mathurinm — APPROVED** [review]
warnings are fixed except for the LegacyVersion one

File: benchopt/cli/main.py (hunk 1 above)
Comment (tomMoral) [reply benchopt/cli/main.py:45]: You need to enable the autocompletion following the first paragraph of this page:
https://benchopt.github.io/cli.html

Maybe a useful feature would be to have these instruction display by running `benchopt config autocomplete`?

**Pull request merged**